### DISCUSSION TITLE
How can you detect indentation and dedentation?

### DISCUSSION THREAD

**Discussion by DeepBhat** [petitparser/dart-petitparser#128] [Q&A] [answered]
I was trying to write a simple python-like grammar using this library. The grammar depends on detecting indentation and dedentation, like how Python does it. I can detect indentation with `\t` but I don't think that is a good idea. What can I do?

**Comment by renggli** [comment, marked as answer]
This is non-trivial, see the corresponding discussions on Stack Overflow: https://stackoverflow.com/questions/4205442/peg-for-python-style-indentation, https://stackoverflow.com/questions/11659095/parse-indentation-level-with-peg-js

I have not tried to do it myself with PetitParser, but know that other people have successfully written grammars for Markdown, Python and the like: http://scg.unibe.ch/research/indentParsing (the examples given here are for the Smalltalk version of this library).

**Comment by renggli** [comment]
With https://github.com/petitparser/dart-petitparser/commit/d03b23bb99c7cf507f8f5f0a3b8dfc6691b2b68d I've committed initial support to make indentation based grammars easier (they are still complicated). The implementation is based on the ideas of https://stackoverflow.com/a/56926044/82303.

**Reply by gmpassos** [reply]
Will be nice to have a simple Python grammar as example.
